enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,170 @@
+---
+title: cli-in-cluster-management
+authors:
+  - "@sallyom"
+reviewers:
+  - TBD
+  - "@soltysh"
+approvers:
+  - TBD
+  - "@soltysh"
+creation-date: 2019-12-03
+last-updated: 2020-03-05
+status: provisional
+---
+
+# CLI In-Cluster Management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal is describing the mechanism for how CLI authors can deliver tools to OpenShift clusters.  An in-cluster feature
+is needed to manage various CLIs available for OpenShift and related services.  
+The goal is for a user to discover, install and upgrade tools easily and from a single
+location.  
+
+Each component is responsible for building its artifacts and registering information regarding supplied binaries
+to a central index location and/or image.  Currently, that location is [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/).
+The CLI in-cluster manager will retrieve binaries from images that package the artifacts from the index.
+Additionally, CLI manager must be supported in disconnected environments.  Currently, we provide disconnected environments 
+cli-artifacts and oc download links that do not require anything outside the cluster.  The goal is to provide the same for other
+binaries/tools/CLIs.  A local image registry with mirrored CLI images can provide binaries in disconnected clusters.      
+
+The CLI in-cluster manager will provide a mechanism to provide, list, install, and upgrade OpenShift tools/plugins/CLIs from
+within a cluster.  The [ConsoleCLIDownload CRD](https://github.com/openshift/api/blob/master/console/v1/0000_10_consoleclidownload.crd.yaml) 
+will be extended to meet the requirements of a new CLI manager.  Each CLI will provide metadata, descriptions, and image references in a Custom Resource.  
+
+An oc subcommand, `oc cli-index`,  will allow users to interact with CLI CRs to list, update, 
```

Comment:
I am not sure if splitting the commands between `cli-index` and `image extract` will be a good user experience for the audience targeted with those tools. Maybe a more integrated command including `image extract` would be better.

Reply:
updated to include krew